### PR TITLE
Fix token status transition in TPS web UI

### PR DESCRIPTION
**Pull request by ckelleyRH**
The XML adaptor was removed so additional work in the JS is needed.

Resolves: #2165098

### CODE REVIEW

**Review by fmarco76 — APPROVED** [review]
Not clear the `input` type and value but the changes look good.

File: base/server/share/webapps/pki/js/pki-ui.js [hunk 1]
```diff
@@ -184,7 +184,12 @@ var Dialog = Backbone.View.extend({
     saveField: function(input) {
         var self = this;
         var name = input.attr("name");
-        var value = input.val();
+        var value;
+        if (name == "status") {
+            value = self.entry.nextStates[input[0].selectedIndex - 1];
+        } else {
+            value = input.val();
+        }
```
Comment (edewata) [inline base/server/share/webapps/pki/js/pki-ui.js:192]: Is `status` a TPS token-specific attribute? In that case you might want to add a `saveField()` in the `TokenDialog` class instead to perform TPS token-specific operation.

File: base/tps/shared/webapps/tps/js/token.js [hunk 2]
```diff
@@ -204,7 +204,7 @@ var TokenPage = EntryPage.extend({
                 if (dialog.entry.status != self.model.attributes.status) {
 
                     self.model.changeStatus({
-                        status: dialog.entry.status,
+                        status: dialog.entry.status.name.name,
```
Comment (edewata) [inline base/tps/shared/webapps/tps/js/token.js:207]: This `dialog` is an instance of `TokenDialog`.

File: base/server/share/webapps/pki/js/pki-ui.js (hunk 1 above)
Comment (ckelleyRH) [reply base/server/share/webapps/pki/js/pki-ui.js]: Yes it maps to the `TokenStatus` class in Java-land - I have created the new `saveField()` method.

**Comment by ckelleyRH** [comment]
> Thanks for the update! I didn't test this, but the changes look good.

Still works on my machine, so I will merge, thanks!

**Pull request merged**